System/UI/Textbox.gd

```diff
@@ -116,82 +229,61 @@ func execute_markup(pack:Pack):
 			else:
 				args = []
 			pack.text = command
-			break
-	match pack.text:
-		"n":
-			$Backdrop/Label.bbcode_text += "\n"
-		"center":
-			if not center:
-				$Backdrop/Label.bbcode_text += "[center]"
-			else:
-				$Backdrop/Label.bbcode_text += "[/center]"
-			center = not center
-		"c":
-			if diffcolor:
-				$Backdrop/Label.bbcode_text += "[/color]"
-			if not args:
-				diffcolor = false
-			else:
-				$Backdrop/Label.bbcode_text += "[color=#"+Colors.string_to_hex(args[0])+"]"
-				diffcolor = true
-		"e":
-			Commands.call_command("emo", main.top_script(), args)
-			#update_emotion(args[0])
-		"$":
-			return main.stack.variables.get_string(args[0])
-		"sfx":
-			pass
-		"sound":
-			pass
-		"delay":   # delay character printing for a time
-			pass
-		"spd":
-			pass
-		"_fullspeed":
-			pass
-		"_endfullspeed":
-			pass
-		"wait":  # set wait mode to auto or manual
-			pass
-		"type":
-			pass
-		"next":
-			queue_free()
-		"tbon":
-			pass
-		"tboff":
-			pass
-		"e":
-			pass
-		"f":
-			pass
-		"s":
-			pass
-		"p":
-			pass
+			pack.args = args
+			return pack
+	print("command not found:", pack.text)
+	return pack
+
+func tokenize_text(text_to_print):
+	var next_pack
+	var packs = []
+	var v = get_next_pack(text_to_print)
+	next_pack = v[0]
+	text_to_print = v[1]
+	while text_to_print:
+		packs.append(next_pack)
+		v = get_next_pack(text_to_print)
+		next_pack = v[0]
+		text_to_print = v[1]
+	packs.append(next_pack)
+	return packs
+
+func _set_speaking_animation(name):
+	for character in Commands.get_speaking_char():
+		character.play_state("talk")
+		
+func strip_bbcode(source:String) -> String:
+	var regex = RegEx.new()
```

Comment:
I'm not a fan of having to do this, but I think it's probably cleaner than bookeeping the length elsewhere so I'm OK with it. On the plus side it makes it easier if we want to support arbitrary bbcode in the future.